### PR TITLE
WIP: Learn how to pin using cid and what to do with file path

### PR DESCRIPTION
**Pull request by cd10012**
This PR is for issue:https://github.com/ipfs/ipfs-cluster/issues/450

Notes: 
https://godoc.org/github.com/ipfs/go-ipfs-path#ParsePath

For the second part of this PR: https://github.com/ipfs/ipfs-cluster/issues/466

### CODE REVIEW

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: a0f5c36d96e17d3b10896269103faac22502556b
  * Invalid signoff. Commit message must end with  
    **License: MIT  
Signed-off-by:**

Code contribution guidelines for IPFS Cluster are available at https://cluster.ipfs.io/developer/contribute/#code-contribution-guidelines

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 2772ad77b598aa130a5863e7b4312cb50e4c1fb5
  * Invalid signoff. Commit message must end with  
    **License: MIT  
Signed-off-by:**

Code contribution guidelines for IPFS Cluster are available at https://cluster.ipfs.io/developer/contribute/#code-contribution-guidelines

---
This message was auto-generated by https://gitcop.com

File: ipfscluster.go [hunk 1]
```diff
@@ -74,6 +74,8 @@ type IPFSConnector interface {
 	ID() (api.IPFSID, error)
 	Pin(context.Context, *cid.Cid, bool) error
 	Unpin(context.Context, *cid.Cid) error
+	PinPath(pinPath string, p api.Pin) error
```
Comment (hsanjuan) [inline ipfscluster.go]: hey, these functions should be in cluster.go as part of the `Cluster` object (main component). The IPFSConnector should just offer something `Resolve(ctx, path string) *cid`.

I think that this `Resolve()` function should have the `IsJustAKey()` logic that you did below. Eventually, if needed, it should use the `doPost()` to perform a request to the IPFS daemon (see how other methods do it), parse the response, extract and decode the Cid and return it.

In `cluster.go` you would have something very similar to:

```
func (c *Cluster) PinPath(path string, pin api.Pin) error {
   ci, err := c.ipfsconn.Resolve(path)
   pin.Cid = ci
   return c.Pin(pin)
}
```

Once you're there, we can discuss the next steps

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 2772ad77b598aa130a5863e7b4312cb50e4c1fb5
  * Invalid signoff. Commit message must end with  
    **License: MIT  
Signed-off-by:**

Code contribution guidelines for IPFS Cluster are available at https://cluster.ipfs.io/developer/contribute/#code-contribution-guidelines

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 2772ad77b598aa130a5863e7b4312cb50e4c1fb5
  * Invalid signoff. Commit message must end with  
    **License: MIT  
Signed-off-by:**

Code contribution guidelines for IPFS Cluster are available at https://cluster.ipfs.io/developer/contribute/#code-contribution-guidelines

---
This message was auto-generated by https://gitcop.com

**Comment by cd10012** [comment]
@hsanjuan I just begun adding the handlers for the `rest/api`. For the function that parses the request, should it use the `Resolve()` method in `ipfsconn` that I wrote? 

I'll have that body response parser also return the `path`. That's when I make the RPC call as written in `pinPathHandler`? Also should `/ipns/<path>` and `/ipfs/<path>` use the same handler? 

Thanks again for the help!

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: c709a862c6c0c85534311906330cc9efad40d32a
  * Your subject line is longer than 80 characters

Code contribution guidelines for IPFS Cluster are available at https://cluster.ipfs.io/developer/contribute/#code-contribution-guidelines

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 0d86bb0674309320da34954ca7b8143932d44ea4
  * Invalid signoff. Commit message must end with  
    **License: MIT  
Signed-off-by:**

Code contribution guidelines for IPFS Cluster are available at https://cluster.ipfs.io/developer/contribute/#code-contribution-guidelines

---
This message was auto-generated by https://gitcop.com

**Comment by hsanjuan** [comment]
Hello, any progress here @cd10012 ?

**Comment by cd10012** [comment]
Apologies for not coming through with this I got stuck; I see that you guys have moved away from this implementation significantly so I'll quietly close this out :(

**Comment by hsanjuan** [comment]
We have not moved away from this. I'll finish it if you can't, no worries.

**Comment by kishansagathiya** [comment]
@hsanjuan Have you started on this? or else I can finish this.

**Comment by hsanjuan** [comment]
@kishansagathiya yes please, take it over

**Comment by kishansagathiya** [comment]
This PR is continued by https://github.com/ipfs/ipfs-cluster/pull/634
Thanks for your contribution @cd10012

**Comment by cd10012** [comment]
@kishansagathiya No, thank you for picking up the rest of this mess!

**Pull request closed without merging**